File: tests/src/main/java/com/github/klee0kai/test/di/house/ToolsModule.java
package com.github.klee0kai.test.di.house;

import com.github.klee0kai.stone.annotations.module.Module;
import com.github.klee0kai.test.house.kitchen.storagearea.Cookware;

@Module
public interface ToolsModule {

    Cookware cookware();

}

File: tests/src/commonMain/kotlin/com/github/klee0kai/test/di/house/simple/ToolsModule.kt
package com.github.klee0kai.test.di.house.simple

import com.github.klee0kai.stone.annotations.module.Module
import com.github.klee0kai.stone.annotations.module.Provide
import com.github.klee0kai.test.house.kitchen.storagearea.Clothes
import com.github.klee0kai.test.house.kitchen.storagearea.Cookware
import com.github.klee0kai.test.house.kitchen.storagearea.Sanitizers

@Module
interface ToolsModule {
    @Provide(cache = Provide.CacheType.Factory)
    fun cookware(): Cookware?

    @Provide(cache = Provide.CacheType.Factory)
    fun clothes(): Clothes?

    @Provide(cache = Provide.CacheType.Factory)
    fun sanitizers(): Sanitizers?
}
